src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,154 @@
+define([
+    // Change this to filer vs. filer.min if you need to debug Filer
+    "thirdparty/filer/dist/filer"
+],function(Filer){
+    "use strict";
+
+    function ProjectStats() {};
+    
+    var _fs = new Filer.FileSystem();
+    var _shell = new _fs.Shell();
+    var _Path = Filer.Path;
+    var _root;
+
+    ProjectStats.getFileSystem = function(){
+        if(_fs){
+            return _fs;
+        }
+    };
+
+    // keep track of project state
+    var _projectInfo = {
+        size: 0,
+        fileCount: 0
+    };
+    // existance of index.html
+    var _hasIndex = false;
+    function _checkIndex(pathname,newValue){
+        if(pathname === _Path.join(_root,"index.html")){
+            _hasIndex = newValue;
+        }
+    }
+
+    //walk the whole file tree and count its size
+    ProjectStats.init = function(root,callback){
+        function addSize(path,next){
+            // If the current path is a directory, do nothing
+            if(path.endsWith("/")) {
+                return next();
+            }
+
+            _fs.stat(path,function(err,stats){
```

Comment:
You always need to deal with the `err` case in your callback.  Also, you should deal with dir vs. file nodes, since you only really care about files.  You *could* keep track of both, and just record `0` for the size of a dir?